packages/qwik-city/middleware/request-handler/resolve-request-handlers.ts

```diff
@@ -407,7 +407,9 @@ export function getPathname(url: URL, trailingSlash: boolean | undefined) {
       url.pathname = url.pathname.slice(0, -1);
     }
   }
-  return url.toString().substring(url.origin.length);
+  // strip internal search params
+  const search = url.search.slice(1).replaceAll(/&?q(action|data|func)=[^&]+/g, '');
```

Comment:
I would still suggest using URLSearchParams and constants instead. But they are in runtime package.

Something like this:

```ts
[QACTION_KEY, QFN_KEY, QDATA_KEY].forEach(key => url.searchParams.delete(key))
```

The browsers and other runtimes have API that makes it easy to work with search params, no need to wrestle with strings.

Reply:
I first had this, but even when it's fully supported it will change the search string a bit. That's why the test doesn't have `=` for the second parameter.

When you remove searchParams, it will convert the rest to search parameters and `?myquery` becomes `?myquery=`. Since we're a framework I want to avoid messing with the URL whenever we can.